### ISSUE TITLE
[BUG] `SIMD.store/load` for `DType.bool`

### ISSUE THREAD

**Issue by rd4com** [modular/modular#3229]
### Bug description

Hello, after implementing `List.__contains__` for `Scalar[DT]`,
tests failed with `DType.bool` !

Minimal repro:
```mojo
alias width = simdwidthof[DType.bool]()
var x = DTypePointer[DType.bool].alloc(width)
alias T = SIMD[DType.bool, width]
T.store(x, T(True))
print(T.load(x)) #32 elements True
x[0] = False
print(T.load(x)) #8 firsts elements False
x.free()
```
Note: it seem to work correctly with others `DType`

Important: if we initialized with `T.store(x, T(False))` instead,
and do `x[0] = True`, it seem to work correctly, but it's not:
assigning to offset 1 results in item 8 being changed

This test shows different results:
```mojo
for i in range(width): 
    print(x[i])`        # see first four bytes
```

Possible fix:
It seem to think that the size of the type is 8 bytes
(debugging with `DTypePointer.__setitem__` and `__getitem__`)

(ping to @laszlokindrat)

### Steps to reproduce

Nothing special, the steps are in the description :+1: 

### System information

```shell
- Ubuntu 23.10
- mojo 2024.7.1105 (211527ca)
- modular 0.7.2 (d0adc668)
```

**Comment by laszlokindrat** [comment]
@rd4com Would you mind trying to reproduce this on the latest nightly? There have been several patches over the past two months that addressed potentially related issues, so this might no longer be a problem.

**Comment by rd4com** [comment]
@laszlokindrat , yes good job with all of this :+1: :heart_on_fire: 

It could be just a small thing, but the bug is still there:
```mojo
def main():
    alias width = 32
    var x = UnsafePointer[Scalar[DType.bool]].alloc(32)
    alias T = SIMD[DType.bool, width]
    x.store(0,T(True))
    print(x.load[width=32]()) #32 elements True
    x[0] = False
    print(x.load[width=32]()) #8 firsts elements False
    x.free()
```
With `DType.uint8`, it seem to work great (no bug) :+1:

**Comment by helehex** [comment]
also see #2813

**Comment by laszlokindrat** [comment]
This still reproduces:
```mojo
def main():
    alias width = 32
    var x = UnsafePointer[Scalar[DType.bool]].alloc(width)
    x.store(0, SIMD[DType.bool, width](fill=True))
    print(x.load[width=width]()) #32 elements True
    x[0] = False
    print(x.load[width=width]()) #8 firsts elements False
    x.free()
```

I also noticed that setting `x[1] = False` will offset by 8 elements. Indeed, `UnsafePointer.__getitem__` uses `offset` under the hood so it is not capable of properly addressing into a simd vector of bits. Then it also sets 8 consecutive elements, which indicates that maybe even SIMD doesn't handle the bool type correctly?

**Comment by rd4com** [comment]
Hello @laszlokindrat ,
i thought it had to do with alignment, but bool look one bit:
```mojo
from memory import bitcast
def main():
    alias width = 32
    var x = UnsafePointer[Scalar[DType.bool]].alloc(width)
    x.store(0, SIMD[DType.bool, width](fill=True))
    print(bitcast[DType.uint8, 32](x.bitcast[UInt256]()[]))
    #[255, 255, 255, 255, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
    # ^ 8 bits * 4 == 32, bool one bit ?
    x.free()
```
We have pack_bits to go from 32 bools to UInt32,
the first 4 bytes is what would it do !

**Comment by rd4com** [comment]
Hello @laszlokindrat , @JoeLoser !
I think the following commit is related to this issue:

* [stdlib] Fix bool SIMD store/load mismatch for width > 1
https://github.com/modular/modular/commit/74f930dbb5a08edea9cb980bdcc807122d76bdb7

Did not try yet, but seem like an fix.

**Comment by JoeLoser** [comment]
Yes, this is fixed with https://github.com/modular/modular/commit/74f930dbb5a08edea9cb980bdcc807122d76bdb7 👍 - thanks for flagging it @rd4com
